tests/helpers/shellGuiTestCaseBase.py

```diff
@@ -159,52 +160,46 @@ def moveMouseFromCenter(self, imgView, coords, modifier=Qt.NoModifier):
     def strokeMouse(
         self,
         imgView: QAbstractScrollArea,
-        start: Union[QPoint, Iterable[int]],
-        end: Union[QPoint, Iterable[int]],
+        topLeft: Union[QPointF, QPoint, Iterable[numbers.Real]],
```

Comment:
naming `topLeft` and `bottomRight` are nice choices for drawing boxes. This function is, however, also used in other contexts. Then I find `start` and `stop` more fitting.
same for `strokeMouseFromCenter`

Reply:
The problem with `start` and `end` is that `end` suggests that the range is start-inclusive end-exclusive (as in Python ranges), whereas `bottomRight` indicates the opposite (start-inclusive end-inclusive). At least that's how `strokeMouse` worked. Besides, top-left and bottom-right are just points in a 2D image view, which is what we want anyway.

Initially, such naming made me think that `strokeMouseFromCenter` was end-exclusive, so I adjusted my test coordinates accordingly, which of course failed the tests, and I spent some time double-checking the coordinates and other code paths.

If you insist on old names, then I would suggest to at least change `strokeMouse` and all test data to end-exclusive (maybe that's a good idea anyway if we're dealing mostly with ROIs).

Reply:
it's funny, to me it never occurred that when stroking the mouse one of the two points would not be reached (as in an "exclusive" end)... `topLeft` and `bottomRight` imply an ordering in space which is confusing here

Reply:
> `topLeft` and `bottomRight` imply an ordering in space which is confusing here

Don't we always move mouse and draw on a 2D image view?

Reply:
think of it as drawing, would you ever expect, given two points, let's call them `start_point` and `end_point`, that any of those points would not be included in the drawn line. The more I think about it the more I am in favor of renaming again, away from `topLeft` and friends

Reply:
> There is an ordering in implementation (curr_point = start_point + point_increment), although I'm not sure whether it's right or not...

if you had a _start-_ and an _end_point this is exactly what you would expect

Reply:
Yes, you are totally right about the ordering! Press-drag-release sequence could be in any of the 8 possible directions (this should be pretty obvious from the beginning, sorry).

> think of it as drawing, would you ever expect, given two points, let's call them `start_point` and `end_point`, that any of those points would _not_ be included in the drawn line.

Maybe... Again, because start-inclusive end-exclusive is a very common way to define an interval. This may be less true while dealing with graphics, but still could be confusing (my guess is this is the reason why Qt moved from start-end to offset-size format).

Slightly off-topic: there is an interesting essay by Djikstra on a very similar issue: https://www.cs.utexas.edu/users/EWD/transcriptions/EWD08xx/EWD831.html

> The more I think about it the more I am in favor of renaming again, away from `topLeft` and friends

Agreed, `topLeft` and friends definitely need to go.

So, maybe rename these parameters to `startPoint` and `endPoint`? I was thinking about literally the same names this night.

P.S. Sorry to bikeshedding here.